From c747f0f7b21b73d6b00654f61bbedba93fd6f2a5 Mon Sep 17 00:00:00 2001
From: "Chew, Chiau Ee" <chiau.ee.chew@intel.com>
Date: Mon, 31 May 2021 18:05:08 +0800
Subject: [PATCH] dts/arm64/intel: enable Agilex SGMII support

This is to enable device tree to support Agilex SGMII..

Upstream-Status: Pending

Signed-off-by: Chew, Chiau Ee <chiau.ee.chew@intel.com>
---
 .../boot/dts/intel/socfpga_agilex_sgmii.dtsi  | 56 +++++++++++++++++++
 .../boot/dts/intel/socfpga_agilex_socdk.dts   |  1 +
 2 files changed, 57 insertions(+)
 create mode 100644 arch/arm64/boot/dts/intel/socfpga_agilex_sgmii.dtsi

diff --git a/arch/arm64/boot/dts/intel/socfpga_agilex_sgmii.dtsi b/arch/arm64/boot/dts/intel/socfpga_agilex_sgmii.dtsi
new file mode 100644
index 000000000000..59dfa3f9e2d1
--- /dev/null
+++ b/arch/arm64/boot/dts/intel/socfpga_agilex_sgmii.dtsi
@@ -0,0 +1,56 @@
+/*
+* Add this piece of dtsi fragment as #include "socfpga_stratix10_sgmii.dtsi"
+* in the file socfpga_stratix10_socdk.dts. Compile it in the kernel along with
+* socfpga_stratix10.dtsi.
+*/
+
+/{
+	soc {
+		clocks {
+			sgmii_1_clk_0: sgmii_1_clk_0 {
+                                compatible = "fixed-clock";
+                                #clock-cells = <0>;
+                                clock-frequency = <125000000>;  /* 125.00 MHz */
+                                clock-output-names = "sgmii_1_clk_0-out_clk";
+                        }; //end sgmii_1_clk_0 (sgmii_1_clk_0)
+
+                        sgmii_1_clk_125: sgmii_1_clk_125 {
+                                compatible = "fixed-clock";
+                                #clock-cells = <0>;
+                                clock-frequency = <125000000>;  /* 125.00 MHz */
+                                clock-output-names = "sgmii_1_clk_125-out_clk";
+                        }; //end sgmii_1_clk_125 (sgmii_1_clk_125)
+		};
+
+		s10_hps_bridges: bridge@80000000 {
+                        compatible = "altr,bridge-18.1", "simple-bus";
+                        reg = <0x80000000 0x60000000>,
+                                <0xf9000000 0x00100000>;
+                        reg-names = "axi_h2f", "axi_h2f_lw";
+                        //clocks = <&iopll_0 1 &clk_0 &dp_0_video_pll 3 &iopll_0 0>;
+                        //clock-names = "h2f_axi_clock", "h2f_lw_axi_clock", "f2sdram0_clock", "f2sdram2_clock";
+                        #address-cells = <2>;
+                        #size-cells = <1>;
+                        ranges = <0x00000000 0x00000000 0x80000000 0x00040000>,
+                                <0x00000001 0x00000000 0xf9003000 0x00000040>,
+                                <0x00000001 0x00000040 0xf9003040 0x00000008>;
+
+                        sgmii_1_gmii2sgmii: phy@100000000 {
+                                compatible = "altr,gmii-to-sgmii-2.0";
+                                reg = <0x00000001 0x00000000 0x00000040>,
+                                        <0x00000001 0x00000040 0x00000008>;
+                                reg-names = "eth_tse_control_port", "gmii_to_sgmii_adapter_avalon_slave";
+                                clocks = <&sgmii_1_clk_0 &gmac1 1 &sgmii_1_clk_125 &sgmii_1_clk_125>;
+                                clock-names = "clock_in", "emac_gtx_clk", "tse_pcs_ref_clk_clock_connection", "tse_rx_cdr_refclk";
+                        }; //end phy@0x100000040 (sgmii_1_gmii2sgmii)
+                };
+
+	};
+};
+
+&gmac1 {
+        altr,gmii-to-sgmii-converter = <&sgmii_1_gmii2sgmii>;
+        altr,gmii_to_sgmii_converter = <&sgmii_1_gmii2sgmii>;
+        phy-mode = "sgmii";
+        status = "okay";
+};
diff --git a/arch/arm64/boot/dts/intel/socfpga_agilex_socdk.dts b/arch/arm64/boot/dts/intel/socfpga_agilex_socdk.dts
index 510c12fc9599..b6347355a92a 100644
--- a/arch/arm64/boot/dts/intel/socfpga_agilex_socdk.dts
+++ b/arch/arm64/boot/dts/intel/socfpga_agilex_socdk.dts
@@ -3,6 +3,7 @@
  * Copyright (C) 2019, Intel Corporation
  */
 #include "socfpga_agilex.dtsi"
+#include "socfpga_agilex_sgmii.dtsi"
 
 / {
 	model = "SoCFPGA Agilex SoCDK";
-- 
2.17.1

